portfolio/src/main/webapp/mocha_tests/calendar_test.js

```diff
@@ -0,0 +1,131 @@
+ * elements of type time or date, the HTML element already checks for 
+ * some invalid input. 
+ * Specifically, the numbers inputted for month, date, hour, and minute
+ * that come from these HTML elements are valid numbers for date and time.
+ * For example, 20 is not a valid month, and 70 is not a valid minute.  
+ * Invalid inputs in this aspect will not be tested in the following tests. 
+ */
+
+describe('#parseTodayString', function () {
+  describe('extra zero padding', function () {
+    it('should return \'2020-01-02\'', function () {
+      const year = '2020';
+      const month = '1';
+      const date = '2'; 
+      assert.equal(calendar._test.parseTodayString(year, month, date), '2020-01-02');
+    });
+  });
+
+  describe('two-digit dates', function () {
+    it('should return \'2020-11-22\'', function () {
+      const year = '2020';
+      const month = '11';
+      const date = '22'; 
+      assert.equal(calendar._test.parseTodayString(year, month, date), '2020-11-22');
+    });
+  });
+});
+
+describe('#getClosestNextHour', function() {
+  describe('next hour is 5pm', function() {
+    it('should return 17:00', function() {
+      const workHourStartString = '09:00';
+      const workHourEndString = '17:00';
+      const nextHour = '17';
+      assert.equal(calendar._test.getClosestNextHour(
+          nextHour, workHourStartString, workHourEndString), '17:00'); 
+    });
+  });
+
+  describe('next hour is 4pm', function() {
+    it('should return 16:00', function() {
+      const workHourStartString = '09:00';
+      const workHourEndString = '17:00';
+      const nextHour = '16';
+      assert.equal(calendar._test.getClosestNextHour(
+          nextHour, workHourStartString, workHourEndString), '16:00'); 
+    });
+  });
+
+  describe('next hour is after working hour ends', function() {
+    it('should return 17:00', function() {
+      const workHourStartString = '09:00';
+      const workHourEndString = '17:00';
+      const nextHour = '20';
+      assert.equal(calendar._test.getClosestNextHour(
+          nextHour, workHourStartString, workHourEndString), '17:00'); 
+    });
+  });
+  
+  describe('next hour is before working hour starts', function() {
+    it('should return 09:00', function() {
+      const workHourStartString = '09:00';
+      const workHourEndString = '17:00';
+      const nextHour = '5';
+      assert.equal(calendar._test.getClosestNextHour(
+          nextHour, workHourStartString, workHourEndString), '09:00'); 
+    });
+  });
+  
+  describe('extra zero padding', function() {
+    it('should return 08:00', function() {
+      const workHourStartString = '06:00';
+      const workHourEndString = '17:00';
+      const nextHour = '8';
+      assert.equal(calendar._test.getClosestNextHour(
+          nextHour, workHourStartString, workHourEndString), '08:00'); 
+    });
+  });
+
+});
+
+describe('#isWorkHourValid', function () {
+  describe('valid working hour', function () {
```

Comment:
I noticed in this pull request that sometimes you say working hour and sometimes you say work hour. Can we try to be consistent and always say working hour?

Reply:
Updated.